scripts/int-log-compiler.sh

```diff
@@ -193,52 +205,65 @@ if [ ! -x "${TEST_BIN}" ]; then
-    sleep 1 # give daemon time to bind to ports
-    PID=$(cat ${SIM_PID_FILE})
-    echo "simulator PID: ${PID}";
-    netstat -ltpn 2> /dev/null | grep "${PID}" | grep -q "${SIM_PORT_DATA}"
-    ret_data=$?
-    netstat -ltpn 2> /dev/null | grep "${PID}" | grep -q "${SIM_PORT_CMD}"
-    ret_cmd=$?
-    if [ \( $ret_data -eq 0 \) -a \( $ret_cmd -eq 0 \) ]; then
-        echo "Simulator with PID ${PID} bound to port ${SIM_PORT_DATA} and " \
-             "${SIM_PORT_CMD} successfully.";
-        break
-    fi
-    echo "Port conflict? Cleaning up PID: ${PID}"
-    kill "${PID}"
-    BACKOFF=$((${BACKOFF}*${BACKOFF_FACTOR}))
-    echo "Failed to start simulator: port ${SIM_PORT_DATA} or " \
-         "${SIM_PORT_CMD} probably in use. Retrying in ${BACKOFF}."
-    sleep ${BACKOFF}
-    if [ $i -eq 10 ]; then
-        echo "Failed to start simulator after $i tries. Giving up.";
-        exit 1
-    fi
-done
+if [ -n "${SIM_BIN}" ]; then
+    SIM_LOG_FILE=${TEST_BIN}_simulator.log
+    SIM_PID_FILE=${TEST_BIN}_simulator.pid
+    SIM_TMP_DIR=$(mktemp --directory --tmpdir=/tmp tpm_server_XXXXXX)
+    PORT_MIN=1024
+    PORT_MAX=65534
+    BACKOFF_FACTOR=2
+    BACKOFF=1
+    for i in $(seq 10); do
+        SIM_PORT_DATA=$(od -A n -N 2 -t u2 /dev/urandom | awk -v min=${PORT_MIN} -v max=${PORT_MAX} '{print ($1 % (max - min)) + min}')
+        SIM_PORT_CMD=$((${SIM_PORT_DATA}+1))
+        echo "Starting simulator on port ${SIM_PORT_DATA}"
+        simulator_start ${SIM_BIN} ${SIM_PORT_DATA} ${SIM_LOG_FILE} ${SIM_PID_FILE} ${SIM_TMP_DIR}
+        sleep 1 # give daemon time to bind to ports
+        PID=$(cat ${SIM_PID_FILE})
+        echo "simulator PID: ${PID}";
+        netstat -ltpn 2> /dev/null | grep "${PID}" | grep -q "${SIM_PORT_DATA}"
+        ret_data=$?
+        netstat -ltpn 2> /dev/null | grep "${PID}" | grep -q "${SIM_PORT_CMD}"
+        ret_cmd=$?
+        if [ \( $ret_data -eq 0 \) -a \( $ret_cmd -eq 0 \) ]; then
+            echo "Simulator with PID ${PID} bound to port ${SIM_PORT_DATA} and " \
+                  "${SIM_PORT_CMD} successfully.";
+            break
+        fi
+        echo "Port conflict? Cleaning up PID: ${PID}"
+        kill "${PID}"
+        BACKOFF=$((${BACKOFF}*${BACKOFF_FACTOR}))
+        echo "Failed to start simulator: port ${SIM_PORT_DATA} or " \
+              "${SIM_PORT_CMD} probably in use. Retrying in ${BACKOFF}."
+        sleep ${BACKOFF}
+        if [ $i -eq 10 ]; then
+            echo "Failed to start simulator after $i tries. Giving up.";
+            exit 1
+        fi
+    done
+else
+    SIM_PORT_DATA=0
+fi
+
 # start an instance of the tpm2-abrmd daemon for the test, use port from above
 TABRMD_LOG_FILE=${TEST_BIN}_tabrmd.log
 TABRMD_PID_FILE=${TEST_BIN}_tabrmd.pid
-TABRMD_NAME=com.intel.tss2.Tabrmd${SIM_PORT_DATA}
+if [ -n "${SIM_BIN}" ]; then
+    TABRMD_NAME=com.intel.tss2.Tabrmd${SIM_PORT_DATA}
+else
+    TABRMD_NAME=com.intel.tss2.Tabrmd
```

Comment:
If you're starting the daemon here with this name on dbus this will have to run as the `tss` user or root? Will this prevent execution as a regular user?

Reply:
Yes. root is required to start up tpm2-abrmd daemon due to the permission of /dev/tpm0 I guess:
```
$ls -l /dev/tpm0 
crw------- 1 tss tss 10, 224 Dec 14 13:39 /dev/tpm0
```